lib/cloud/test/redshift_serverless.go

```diff
@@ -0,0 +1,136 @@
+	if m.TagsByARN == nil {
+		return &redshiftserverless.ListTagsForResourceOutput{}, nil
+	}
+	return &redshiftserverless.ListTagsForResourceOutput{
+		Tags: m.TagsByARN[aws.StringValue(input.ResourceArn)],
+	}, nil
+}
+func (m RedshiftServerlessMock) GetCredentialsWithContext(aws.Context, *redshiftserverless.GetCredentialsInput, ...request.Option) (*redshiftserverless.GetCredentialsOutput, error) {
+	if m.GetCredentialsOutput == nil {
+		return nil, trace.AccessDenied("access denied")
+	}
+	return m.GetCredentialsOutput, nil
+}
+
+// RedshiftServerlessWorkgroup returns a sample redshiftserverless.Workgroup.
+func RedshiftServerlessWorkgroup(name, region string) *redshiftserverless.Workgroup {
+	return &redshiftserverless.Workgroup{
+		BaseCapacity: aws.Int64(32),
+		ConfigParameters: []*redshiftserverless.ConfigParameter{{
+			ParameterKey:   aws.String("max_query_execution_time"),
+			ParameterValue: aws.String("14400"),
+		}},
+		CreationDate: aws.Time(sampleTime),
+		Endpoint: &redshiftserverless.Endpoint{
+			Address: aws.String(fmt.Sprintf("%v.1234567890.%v.redshift-serverless.amazonaws.com", name, region)),
+			Port:    aws.Int64(5439),
+			VpcEndpoints: []*redshiftserverless.VpcEndpoint{{
+				VpcEndpointId: aws.String("vpc-endpoint-id"),
+				VpcId:         aws.String("vpc-id"),
+			}},
+		},
+		NamespaceName:      aws.String("my-namespace"),
+		PubliclyAccessible: aws.Bool(true),
+		Status:             aws.String("AVAILABLE"),
+		WorkgroupArn:       aws.String(fmt.Sprintf("arn:aws:redshift-serverless:%v:1234567890:workgroup/some-uuid-for-%v", region, name)),
+		WorkgroupId:        aws.String(fmt.Sprintf("some-uuid-for-%v", name)),
+		WorkgroupName:      aws.String(name),
+	}
+}
+
+// RedshiftServerlessEndpointAccess returns a sample redshiftserverless.EndpointAccess.
+func RedshiftServerlessEndpointAccess(workgroup *redshiftserverless.Workgroup, name, region string) *redshiftserverless.EndpointAccess {
+	return &redshiftserverless.EndpointAccess{
+		Address:            aws.String(fmt.Sprintf("%s-endpoint-xxxyyyzzz.1234567890.%s.redshift-serverless.amazonaws.com", name, region)),
+		EndpointArn:        aws.String(fmt.Sprintf("arn:aws:redshift-serverless:%s:1234567890:managedvpcendpoint/some-uuid-for-%v", region, name)),
+		EndpointCreateTime: aws.Time(sampleTime),
+		EndpointName:       aws.String(name),
+		EndpointStatus:     aws.String("AVAILABLE"),
+		Port:               aws.Int64(5439),
+		VpcEndpoint: &redshiftserverless.VpcEndpoint{
+			VpcEndpointId: aws.String("vpce-id"),
+			VpcId:         aws.String("vpc-id"),
+		},
+		WorkgroupName: workgroup.WorkgroupName,
+	}
+}
+
+// RedshiftServerlessGetCredentialsOutput return a sample redshiftserverless.GetCredentialsOutput.
+func RedshiftServerlessGetCredentialsOutput(user, password string) *redshiftserverless.GetCredentialsOutput {
```

Comment:
nit: Receive a clock parameter ([`clockwork.Clock`](https://pkg.go.dev/github.com/jonboulle/clockwork#Clock)) instead of fixing it with `sampleTime`.